presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -1048,29 +1049,38 @@ public void testVirtualBucketing()
         }
     }
 
-    // TODO: plan verification https://github.com/prestodb/presto/issues/16031
     // TODO: enable all materialized view tests after https://github.com/prestodb/presto/pull/15996
     @Test
     public void testMaterializedViewOptimization()
     {
         QueryRunner queryRunner = getQueryRunner();
+        String table = "orders_partitioned";
+        String view = "test_orders_view";
         try {
-            queryRunner.execute("CREATE TABLE orders_partitioned WITH (partitioned_by = ARRAY['ds']) AS " +
+            queryRunner.execute(format("CREATE TABLE %s WITH (partitioned_by = ARRAY['ds']) AS " +
                     "SELECT orderkey, orderpriority, '2020-01-01' as ds FROM orders WHERE orderkey < 1000 " +
                     "UNION ALL " +
-                    "SELECT orderkey, orderpriority, '2019-01-02' as ds FROM orders WHERE orderkey > 1000");
-
-            assertUpdate("CREATE MATERIALIZED VIEW test_orders_view WITH (partitioned_by = ARRAY['ds']) " +
-                    "AS SELECT orderkey, orderpriority, ds FROM orders_partitioned");
+                    "SELECT orderkey, orderpriority, '2019-01-02' as ds FROM orders WHERE orderkey > 1000", table));
+            assertUpdate(format("CREATE MATERIALIZED VIEW %s WITH (partitioned_by = ARRAY['ds']) " +
+                    "AS SELECT orderkey, orderpriority, ds FROM %s", view, table));
             assertTrue(getQueryRunner().tableExists(getSession(), "test_orders_view"));
             assertUpdate("REFRESH MATERIALIZED VIEW test_orders_view WHERE ds='2020-01-01'", 255);
 
-            String viewQuery = "SELECT orderkey from test_orders_view where orderkey <  10000 ORDER BY orderkey";
-            String baseQuery = "SELECT orderkey from orders_partitioned where orderkey <  10000 ORDER BY orderkey";
+            String viewQuery = format("SELECT orderkey from %s where orderkey < 10000 ORDER BY orderkey", view);
+            String baseQuery = format("SELECT orderkey from %s where orderkey < 10000 ORDER BY orderkey", table);
 
             MaterializedResult viewTable = computeActual(viewQuery);
             MaterializedResult baseTable = computeActual(baseQuery);
             assertEquals(viewTable, baseTable);
+
+            assertPlan(getSession(), viewQuery, anyTree(
+                    filter("orderkey < BIGINT'10000'", PlanMatchPattern.constrainedTableScan(table,
+                                    ImmutableMap.of("ds", singleValue(createVarcharType(10), utf8Slice("2019-01-02"))),
+                                    ImmutableMap.of("orderkey", "orderkey"))),
+                    filter("orderkey_17 < BIGINT'10000'", PlanMatchPattern.constrainedTableScan(view, ImmutableMap.of(), ImmutableMap.of("orderkey_17", "orderkey")))));
```

Comment:
Even though our Plan node id generator is sequential, I think these variables may not be very reliable. Do we have a way to assert it without relying on these variables?

Reply:
The id here actually doesn't matter. It can be `orderkey_0`, `orderkey_1` or any random string as long as it 1) gets mapped to `orderkey` in `columnReferences` 2) uses an alias different from the one already been used in line1077.

I chose `orderkey_17` so it could be aligned with the actual query plan.

Reply:
Oh wonderful! thanks for explaining!